From 72bd28ac1334a8721ae4613b73345a527a508d20 Mon Sep 17 00:00:00 2001
From: Stefan Haller <stefan@haller-berlin.de>
Date: Mon, 12 Oct 2020 17:13:51 +0200
Subject: [PATCH 09/12] git-gui: Use TCLTK_PATH for the Wish exectuable

This allows us to use a non-framework version of Tcl/Tk.

This probably doesn't work if you don't pass TCLTK_PATH, but I'm not interested
in fixing this.
---
 git-gui/Makefile | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/git-gui/Makefile b/git-gui/Makefile
index f10caedaa7..0f5ca74b3c 100644
--- a/git-gui/Makefile
+++ b/git-gui/Makefile
@@ -180,7 +180,7 @@ git-gui: GIT-VERSION-FILE GIT-GUI-VARS
 	echo '	'echo \'git-gui version '$(GITGUI_VERSION)'\' >>$@+ && \
 	echo else >>$@+ && \
 	echo '	libdir="$${GIT_GUI_LIB_DIR:-$(libdir_SQ)}"' >>$@+ && \
-	echo '	'exec \"'$$libdir/Git Gui.app/Contents/MacOS/$(subst \,,$(TKEXECUTABLE))'\" \
+	echo '	'exec \"'$$libdir/Git Gui.app/Contents/MacOS/Wish'\" \
 		'"$$0" "$$@"' >>$@+ && \
 	echo fi >>$@+ && \
 	chmod +x $@+ && \
@@ -190,15 +190,15 @@ Git\ Gui.app: GIT-VERSION-FILE GIT-GUI-VARS \
 		macosx/Info.plist \
 		macosx/git-gui.icns \
 		macosx/AppMain.tcl \
-		$(TKFRAMEWORK)/Contents/MacOS/$(TKEXECUTABLE)
+		$(TCLTK_PATH)
 	$(QUIET_GEN)rm -rf '$@' '$@'+ && \
 	mkdir -p '$@'+/Contents/MacOS && \
 	mkdir -p '$@'+/Contents/Resources/Scripts && \
-	cp '$(subst ','\'',$(subst \,,$(TKFRAMEWORK)/Contents/MacOS/$(TKEXECUTABLE)))' \
-		'$@'+/Contents/MacOS && \
+	cp '$(subst ','\'',$(subst \,,$(TCLTK_PATH)))' \
+		'$@'+/Contents/MacOS/Wish && \
 	cp macosx/git-gui.icns '$@'+/Contents/Resources && \
 	sed -e 's/@@GITGUI_VERSION@@/$(GITGUI_VERSION)/g' \
-		-e 's/@@GITGUI_TKEXECUTABLE@@/$(TKEXECUTABLE)/g' \
+		-e 's/@@GITGUI_TKEXECUTABLE@@/Wish/g' \
 		macosx/Info.plist \
 		>'$@'+/Contents/Info.plist && \
 	sed -e 's|@@gitexecdir@@|$(gitexecdir_SQ)|' \
-- 
2.29.0.12.g8e3af0896f.dirty

